src/ppx_import.ml

```diff
@@ -499,18 +516,28 @@ let module_type ~tool_name mapper modtype_decl =
               in
               {modtype_decl with pmty_desc = Pmty_signature psig}
             | {mtd_type = None; _} ->
-              raise_errorf ~loc "Imported module is abstract"
-            | _ -> raise_errorf ~loc "Imported module is indirectly defined")
-    | _ -> raise_errorf ~loc "Invalid [%%import] syntax" )
-  | _ -> default_mapper.module_type mapper modtype_decl
+              Location.raise_errorf ~loc "Imported module is abstract"
+            | _ ->
+              Location.raise_errorf ~loc "Imported module is indirectly defined")
+    | _ -> Location.raise_errorf ~loc "Invalid [%%import] syntax" )
+  | _ -> modtype_decl
+
+let mapper =
+  let open Ppxlib in
+  object
+    inherit [Expansion_context.Base.t] Ast_traverse.map_with_context as super
+
+    method! module_type ctxt modtype_decl =
+      let tool_name = Ppxlib.Expansion_context.Base.tool_name ctxt in
+      let modtype_decl = super#module_type ctxt modtype_decl in
+      module_type ~tool_name modtype_decl
+
+    method! type_declaration ctxt type_decl =
+      let tool_name = Ppxlib.Expansion_context.Base.tool_name ctxt in
+      let type_decl = super#type_declaration ctxt type_decl in
+      type_declaration ~tool_name type_decl
+  end
 
 let () =
-  let open Migrate_parsetree in
-  (* Position 0 is the default, ppx_import should run pretty early,
-     thus using -10 *)
-  Driver.register ~name:"ppx_import" ~args:[] ~position:(-10) Versions.ocaml_411
-    (fun config _cookies ->
-      let tool_name = config.tool_name in
-      let type_declaration = type_declaration ~tool_name in
-      let module_type = module_type ~tool_name in
-      {default_mapper with type_declaration; module_type})
+  Ppxlib.Driver.V2.register_transformation ~rules:[] ~impl:mapper#structure
+    "ppx_import"
```

Comment:
Sorry, somehow I didn't sent this in the review. See the [general comment of my review](https://github.com/ocaml-ppx/ppx_import/pull/54#pullrequestreview-686155575) for more information.

Of course, this is only meant as a possibility for a temporary hack to make sure that `ppx_import` is applied before other rewriters (such as `ppx_sexp_conv`) are.

```suggestion
  let open Ppxlib.Driver in
  (* Currently, ppxlib only provides a way to specify when a rewriter should be applied
     (with respect to other rewriters), if the rewriter is an instrumentation. *)
  let temporary_hack =
    Instrument.V2.make mapper#structure ~position:Instrument.Before
  in
  V2.register_transformation ~rules:[] ~instrument:temporary_hack "ppx_import"
```

Reply:
The hack works, thanks a lot!

Reply:
Cool! Do you want to merge this like that? Or do you want to wait until in ppxlib we've implemented both a way to register `ppx_import` as extension point and a solution to this problem?

Reply:
I don't mind following what you think it is best, @tatchi WDYT?

Reply:
Hey, to come back to this and sum up the situation: the other PPXs apart from `ppx_import` pretty much all already rely on modern ppxlib (and therefore, in their newest version, are incompatible with current `ppx_import` due to relying on OMP1). This port seems very nice, functional and non-intrusive. I've suggested reducing the use of `compiler-libs` at a few places which could be done without much effort, but that's not a huge problem and it's up to you two. Apart from that the only problems this port has are due to missing features in `ppxlib` and improving them would take time:
- it's written as whole file transformation and therefore takes up one whole AST pass by its own. But that's also the case with the current OMP1 implementation.
- it has to register `ppx_import` as "instrumentation", which isn't great but practically doesn't do any harm.

So I think merging and releasing this approach would be a good idea. The two points above can still be improved in a different PR when `ppxlib` is ready for that. But of course, that's just my opinion. Let's see what @tatchi says.

Reply:
Indeed, I'm happy to move ahead as you propose, we can wait a bit to see if @tatchi could address your excellent comments, or if it is busy we can indeed do a release with the current PR and hopefully address the comments later.